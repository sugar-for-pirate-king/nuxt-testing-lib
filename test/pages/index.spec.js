import { render, fireEvent } from '@testing-library/vue'
import PagesIndex from '~/pages/index.vue'

describe('Pages index page spec', () => {
  test('it has feature increment stuff', async () => {
    const { getByText } = render(PagesIndex)
    getByText('Hello, world')
    const button = getByText('Increment')
    await fireEvent.click(button)
    await fireEvent.click(button)
    getByText('You already click the button 2 times')
  })

  test('working good with emitting stuff', async () => {
    const { getByText, getByLabelText } = render(PagesIndex)
    const usernameInput = getByLabelText(/username/i)
    await fireEvent.update(usernameInput, 'Kokomi')
    await fireEvent.keyUp(usernameInput)
    getByText('Hello, Kokomi')
  })
})
