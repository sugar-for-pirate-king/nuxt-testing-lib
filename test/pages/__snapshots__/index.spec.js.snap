// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages index page spec testing router stuff <3 1`] = `[Function]`;
